app/components/receive-menu.js

```diff
@@ -27,17 +36,58 @@ export default Ember.Component.extend({
     toggle(hidden) {
       this.set("hidden", hidden);
     },
+
     missing() {
+      if(this.get("isFirstReceivingPackage")) {
+        this.confirmReceiving(() => this.send("missingPackage"));
+      } else {
+        this.send("missingPackage");
+      }
+    },
+
+    receive() {
+      if(this.get("isFirstReceivingPackage")) {
+        this.confirmReceiving(() => this.send("receivePackage"));
+      } else {
+        this.send("receivePackage");
+      }
+    },
+
+    missingPackage() {
       this.updatePackage(p => {
         p.set("state", "missing");
         p.set("state_event", "mark_missing");
       });
     },
-    receive() {
+
+    receivePackage() {
       this.updatePackage(p => {
         p.set("state", "received");
         p.set("state_event", "mark_received");
       });
-    }
+    },
+  },
+
+  confirmReceiving: function(successCallback) {
+    var _this = this;
+    Ember.$("#confirmReceivingModal").removeClass("open");
+
+    Ember.$("#confirmReceivingModal").foundation("reveal", "open");
+    Ember.$(".loading-indicator").remove();
+
+    Ember.$("#confirmReceivingModal .closeLink").click(() => {
+      _this.closeConfirmBox();
+    });
+
+    Ember.$("#confirmReceivingModal .confirmLink").click(() => {
+      _this.closeConfirmBox();
+      successCallback();
+    });
+  },
+
+  closeConfirmBox: function() {
+    Ember.run.next(function() {
+      Ember.$("#confirmReceivingModal").foundation("reveal", "close");
+    });
```

Comment:
I'm seeing a lot of `Ember.$` here and that makes me worry a bit. Should we be doing things the 'Ember Way' (tm) and using properties to update the component state. It seems that if we keep going directly to jQuery then we bypass Ember's run loop and eventually we'll have inconsistencies down the line. Perhaps actions like clicking a link make sense for jQuery (though I'm not an expert) but shouldn't `removeClass` and `remove()` be set by a property?